DEPLOYMENT.md

```diff
@@ -31,23 +31,61 @@ AWS EC2 Example Configuration:
 # Required Software - System Setup
 Install the following packages:
 ```
-sudo apt update && sudo apt upgrade -y
-sudo apt install -y docker.io docker-compose git python3-pip
+sudo apt update && sudo apt upgrade -y    # refreshes the package index and upgrades any out-of-date base packages, which is crucial for installing Docker/NVIDIA tooling.
+sudo apt install -y docker.io docker-compose git python3-pip 
 sudo apt install -y nvidia-driver nvidia-container-runtime  # For GPU-based services
 ```
+- nvidia-driver is needed so that the server can communicate with NVIDIA GPUs. Without this, GPU-accelerated preprocessors (like segmentation or local LLMs) won’t run.
+- nvidia-container-runtime integrates NVIDIA GPUs into Docker. Without this, Compose won’t recognize your GPUs.
 
 Post-install steps:
 ```
-sudo usermod -aG docker $USER
-newgrp docker
+sudo usermod -aG docker $USER    # By default, only root can run Docker. Adding your user to the docker group allows you to run Docker commands without sudo
+newgrp docker    # Refreshes, so you don’t need to log out/in before using Docker as a non-root user.
 reboot  # Required after NVIDIA driver install
 ```
 
 Verify:
-```
-docker --version
-docker-compose --version
-nvidia-smi  # For GPU instances
+
+`docker --version`
+
+`docker-compose --version`
+
+`nvidia-smi  # For GPU instances. This should display driver version and usage, similar to:`
```

Comment:
Not sure we need the example below. this can be easily found online, and is not IMAGE-specific.